opentelemetry-sdk/src/logs/record.rs

```diff
@@ -124,6 +125,76 @@ impl LogRecord {
             .flatten()
             .any(|(k, v)| k == key && v == value)
     }
+
+    /// Updates the first occurrence of an attribute with the specified key.
+    ///
+    /// This method searches for the first occurrence of the attribute with the given key
+    /// in the `attributes` collection. If the key is found, its value is updated with the
+    /// provided value. If the key is not found, the attribute is added.
+    ///
+    /// # Arguments
+    ///
+    /// - `key`: A reference to the key of the attribute to update.
+    /// - `value`: A reference to the new value for the attribute.
+    ///
+    /// # Returns
+    ///
+    /// - `Some(AnyValue)`: The old value of the attribute if found and updated.
+    /// - `None`: If the attribute was not found, and a new one was added.
+    ///
+    pub fn update_attribute(&mut self, key: &Key, value: &AnyValue) -> Option<AnyValue> {
+        // First, search for the attribute mutably
+        if let Some(attr) = self
+            .attributes
+            .iter_mut()
+            .find(|opt| opt.as_ref().map(|(k, _)| k == key).unwrap_or(false))
+        {
+            // Take the old value and update the attribute
+            let old_value = attr.take().map(|(_, v)| v);
+            *attr = Some((key.clone(), value.clone()));
+            return old_value;
+        }
+
+        // If not found, add a new attribute
+        self.add_attribute(key.clone(), value.clone());
+        None
+    }
+
+    /// Deletes the first occurrence of an attribute with the specified key.
```

Comment:
should it delete all occurrences instead of first? I understand that we don't de-dup, so if a user believes they accidentally put a PII/something into LogRecord, and if they ended up with duplicates too, the expectation when one calls delete is that that key/value is completely eliminated.

Reply:
I though delete first provides flexibility, and can be called repeatedly to delete all. But I understand, difficult to get use-case where user would like to delete specific/first entry. Have update it to delete all occurrences. Also renamed it to `remove_attribute`, to be consistent with `remove` method of Vector and HashMap.